### DISCUSSION TITLE
Docs Help: Noobs Building from Source and Running Transmission

### DISCUSSION THREAD

**Discussion by komali2** [transmission/transmission#4052] [General]
### What is the issue?

Hello,

I am a noob, and I read a lot of instructions online, including on transmission's website, about installing Transmission. For my platform, Ubuntu 22.04, it was recommended I do 

`sudo apt install transmission` 

See: https://transmissionbt.com/download

However, I've noticed over the last month or so of using Transmission, the RAM usage goes up linearly over the course of days, forcing me to restart. Smells like a memory leak. I googled, and was brought to an issue here https://github.com/transmission/transmission/issues/3077 indicating that the version I have is "several hundred commits out of date." Apparently, the build on ubuntu releases is very out of date?

The recommendation is "to use a nightly build." I would like to do so, and earnestly read the github, READMEs, documentation, and guides elsewhere. I'm led to understand that my steps are as follows:

1. Download a tar from somewhere, I choose the transmission website:https://github.com/transmission/transmission-releases/raw/master/transmission-4.0.0-beta.1%2Br98cf7d9b3c.tar.xz (warning: download link)
2. Install a bunch of dependencies (see https://github.com/transmission/transmission/blob/main/docs/Building-Transmission.md)
3. Unzip the tar

Here I get confused, as the guides in the repo seem mismatched. The above linked on talks about making a `build` directory ONLY if building from git, the `cmake` (from tar) steps don't have that. However, the root readme: https://github.com/transmission/transmission has steps about making build directory, doing a `cmake` command, etc. I choose the root repo README instructions

4. Do a `cmake` command
5. do `make`
6. do `sudo make install`

Now, as a noob, I have completed the steps, and transmission "is built." But I'm pretty confused where to go from here, and I've been extensively googling (including reading the arch linux transmission guide) and still not sure what to do.

I can run the `transmission-daemon` : `cd build/daemon` `./transmission-daemon` but I'm not sure how to make it a service like the arch linux docs say, and anyway, it eventually fails with `Segmentation fault (core dumped)` (I haven't investigated further yet)

I'm used to running transmission by just running a transmission UI, which correctly takes my configs from `~/.config/transmission/` and sees all my in progress torrents and what not. I have a lot of programs pointing at transmission's webserver so I hope to maintain these settings one way or another. However, there is no "transmission" directory in my "build" directory after `make install` ing, so I don't know how to use a 4.0 version of `transmission` UI. I would be happy to use transmission-daemon as a service, but again, am really not sure how or where to find guides on how to do so (I also did some preliminary googling on in general how to make things into a service, but the scope was just too wide with lots of different methods and info, and I was afraid of doing the wrong thing)

I actually love writing documentation, my coworkers hate me for it because it takes me longer to get things done for all my code comments and whatnot, so I'm very happy to contribute on this front, I would just need some guidance first. If someone could help me with some bare-bones direction of where to go from here, I'm more than willing to be a guinea pig and flesh out "for dummies" (like me) instructions. 

### Which version of Transmission?

transmission-4.0.0-beta.1.dev+rcdf817f2e7

**Comment by fredo-47** [comment]
After doing the steps you did, you should be able to use transmission in the same ways as before.

In particular, doing the "sudo make install" step, the relevant files are copied from the build directory to your system's path.
I.e., you do not execute them from within the build directory. Instead, because they are now in e.g. "/usr/local/bin" (which is in your PATH variable), you can execute them directly without giving a path.

The daemon can be connected to via web interface, or via third party GUIs. And the GTK and/or QT app can be started e.g. from the terminal or from your application menu 
This has not changed compared to prior versions of transmission.

**Reply by GaryElshaw** [reply]
I think @komali2 might have been using one of the pre-built packages?

**Reply by komali2** [reply]
I'm not sure if I was using a pre build package, i did have steps with `make` and `sudo make install`

**Comment by GaryElshaw** [comment]
As @fredo-47 said, you should be good to go. From here, i would open a terminal window and type: 'transmission-gtk' Or, if you prefer, 'transmission-qt'. Once running, you can pin the application to your dash and not worry about the command line (until next time).

**Reply by komali2** [reply]
I tried launching from dash, but the version shows as 3. Lemme try real quick your other recommendations. Thanks for your replies!

**Reply by komali2** [reply]
When I try launching from application in dash, under "About" it shows version as `3.00 (bb6b5a062e)`

Same for terminal launching `transmission-gtk`

`transmission-qt` isn't an option for me

I'm trying now with `transmission-daemon` from the transmisson-4.0.0 directory to see if I can see version from the web UI. I don't have any GTK programs that i can see for transmission 4 here.

Regarding above, I had expected not to see version 3.00, not version 4.00.

**Reply by GaryElshaw** [reply]
Something has definitely happened between the build and install process. I would try uninstalling everything using the add/remove app for a start, and then having a look in your /usr/local/bin directory to see if any there are any remnants.

**Reply by komali2** [reply]
Ok, actually, it appears version 4 is properly installed in `/usr/bin`, just various desktop files or whatever aren't updated, nor is there a transmission-gtk installed for version 4, but that's fine, I wanted the daemon anyway.

Thanks for all the help!

**Comment by aikudinov** [comment]
It seems to be caused by ubuntu/debian openssl3 patch. I reverted it and used gentoo openssl3 patch, transmission is under 122Mb for days now instead of gigabytes.

https://gitweb.gentoo.org/repo/gentoo.git/tree/net-p2p/transmission/files/transmission-3.00-openssl-3.patch

You can try this ppa: https://launchpad.net/~elhana/+archive/ubuntu/ppa-elhana
